ext/intl/formatter/formatter_format.c

```diff
@@ -103,6 +131,24 @@ PHP_FUNCTION( numfmt_format )
 			}
 			INTL_METHOD_CHECK_STATUS( nfo, "Number formatting failed" );
 			break;
+
+		case FORMAT_TYPE_DECIMAL:
+			if (!try_convert_to_string(number)) {
+				RETURN_THROWS();
+			}
```

Comment:
This _could_ be simplified as at this point the value should be ``int|float|string`` and integers and floats always have string representations.